spec/system/admin/admin_visits_overview_spec.rb

```diff
@@ -1,14 +1,54 @@
 require "rails_helper"
 
 RSpec.describe "Admin visits overview page", type: :system, js: true do
+  include ActiveSupport::Testing::TimeHelpers
+
   let(:super_admin) { create(:user, :super_admin) }
 
   before do
     sign_in super_admin
     visit admin_path
   end
 
-  it "has analytics" do
-    expect(page.body).to include "Analytics and trends"
+  describe "analytics" do
+    before do
+      travel_to Time.zone.local(2022, 3, 30)
+    end
+
+    it "has analytics" do
+      expect(page.body).to include "Analytics and trends"
+    end
+
+    it "starts with yesterday" do
+      expect(page.body).to include "Yesterday"
+    end
+
+    it "starts a week before" do
+      expect(page.body).to include "Mar 24"
+    end
+
+    it "displays correct number of posts from past week" do
+      create(:article, published_at: Time.zone.local(2022, 3, 30))
```

Comment:
I added these as variables at the top, as part of `describe "analytics"`, but for some reason that didn't work and it found 0 articles inside my `it` blocks. Any idea how to group these at the top, so I can avoid creating them inside each example?

Reply:
You may either want to add the `create` calls to the `before` block, or via a `let!` binding. Create calls outside of an it block (and outside of a before block) won't be run during test execution.

Reply:
Totally forgot the `!` on my `let`s - it's why it wasn't working 🤦🏻‍♀️